grails-app/services/com/unifina/service/SubscriptionService.groovy

```diff
@@ -14,6 +14,16 @@ import grails.compiler.GrailsCompileStatic
 class SubscriptionService {
 	PermissionService permissionService
 
+	void deleteProduct(Product product) {
+		def user = SecUser.where {
+			username == product.owner
+		}.find()
+		getSubscriptionsOfUser(user).toArray().each { Subscription subscription ->
```

Comment:
If I understand correctly, this is deleting the subscriptions of the "troll user".  What I think we should be doing, is deleting subscriptions of the Product.

So instead of `getSubscriptionsOfUser(user)`, something like
`List<Subscription> subscriptions = Subscription.findAllByProduct(product)` ?